subsys/usb/device_next/class/usbd_uvc.c

```diff
@@ -0,0 +1,2268 @@
+/*
+ * Copyright (c) 2025 tinyVision.ai Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_uvc_device
+
+#include <stdlib.h>
+
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/usb_ch9.h>
+#include <zephyr/drivers/usb/udc.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/drivers/video-controls.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/usb/usb_ch9.h>
+#include <zephyr/usb/class/usbd_uvc.h>
+
+#include "usbd_uvc.h"
+#include "../../../drivers/video/video_ctrls.h"
+#include "../../../drivers/video/video_device.h"
```

Comment:
Maybe the definitions from those should be moved to a globally-accessible header? Or if it is already the case, including those by absolute paths would look better imo

Reply:
These are headers meant for driver authors and not an user API. For instance [`udc_comon.h`](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/usb/udc/udc_common.h) needs the same kind of workaround for out-of-tree drivers.

USB Classes sometimes implement drivers interfaces (like CDC ACM), convenient but causes problems... We may consider if we really want a video driver interface for UVC...